Update register results and log warnings

eatkins opened the pull request:
On osx, there can be situations in which the OS fails to register a
watch stream which can break functionality. When this happens, we want
to return an error to the caller and log a warning since this is very
likely to break the application. There is an argument that we should
actually bomb out on an error rather than just log a warning, but that
seems a bit aggressive.

eatkins committed:
Update register results and log warnings

On osx, there can be situations in which the OS fails to register a
watch stream which can break functionality. When this happens, we want
to return an error to the caller and log a warning since this is very
likely to break the application. There is an argument that we should
actually bomb out on an error rather than just log a warning, but that
seems a bit aggressive.

eatkins commented:
see #145. @pvid does this make sense to you?

eatkins committed:
reduce jacoco class threshold

The new CreateStreamException is difficult to induce in a
test situation.

pvid commented on files/jvm/src/main/java/com/swoval/files/FileCacheDirectoryTree.java:
```diff
@@ -414,7 +415,12 @@ public void close() {
     if (directoryRegistry.addDirectory(absolutePath, maxDepth) && directories.lock()) {
       try {
-        watcher.register(absolutePath, maxDepth);
+        final Either<IOException, Boolean> res = watcher.register(absolutePath, maxDepth);
```
is it OK to swallow the exception here? As a user, I'd expect this method to throw if any failure occurred during registration given the method signature

eatkins replied:
That's a good point. It probably should throw.

pvid commented:
> see #145. @pvid does this make sense to you?

🎉  thank you for taking the time to look into this! I left one question, but the changes seem to address the mail issue with the error reporting

eatkins committed:
Throw in FileCacheDirectoryTree on registration failure

The register method of FileCacheDirectoryTree has throws IOException in
the signature so we shouldn't swallow the IOException and should instead
rethrow after logging.
